src/solver/optimisation/opt_appel_solveur_lineaire.cpp

```diff
@@ -274,17 +283,54 @@ bool OPT_AppelDuSimplexe(const OptimizationOptions& options,
             {
                 logs.info() << " solver: resetting";
             }
-            ProbSpx = nullptr;
-            solver = nullptr;
-            PremierPassage = false;
-            goto RESOLUTION;
+            return returnValues;
         }
 
         else
         {
             throw FatalError("Internal error: insufficient memory");
         }
     }
+    returnValues.simplexReturnValue = true;
+    return returnValues;
+}
+
+bool OPT_AppelDuSimplexe(const OptimizationOptions& options,
+                         PROBLEME_HEBDO* problemeHebdo,
+                         int NumIntervalle,
+                         const int optimizationNumber,
+                         const OptPeriodStringGenerator& optPeriodStringGenerator,
+                         IResultWriter& writer)
+{
+    PROBLEME_ANTARES_A_RESOUDRE* ProblemeAResoudre = problemeHebdo->ProblemeAResoudre.get();
+    Optimization::PROBLEME_SIMPLEXE_NOMME Probleme(ProblemeAResoudre->NomDesVariables,
+                                                   ProblemeAResoudre->NomDesContraintes,
+                                                   ProblemeAResoudre->StatutDesVariables,
+                                                   ProblemeAResoudre->StatutDesContraintes);
+
+    auto solver = (MPSolver*)(ProblemeAResoudre->ProblemesSpx[(int)NumIntervalle]);
+
+    mpsWriterFactory mps_writer_factory(problemeHebdo->ExportMPS,
+                                        problemeHebdo->exportMPSOnError,
+                                        optimizationNumber,
+                                        &Probleme,
+                                        options.useOrtools,
+                                        solver);
+
+    bool PremierPassage = true;
+
+    struct returnSimplexSuccessAndTime ret =
+        OPT_TryToCallSimplex(options, problemeHebdo, Probleme, NumIntervalle, optimizationNumber,
+                optPeriodStringGenerator, PremierPassage, writer, mps_writer_factory);
+
+    if (!ret.simplexReturnValue)
+    {
+        PremierPassage = false;
+        ret = OPT_TryToCallSimplex(options, problemeHebdo, Probleme,  NumIntervalle, optimizationNumber,
+                optPeriodStringGenerator, PremierPassage, writer, mps_writer_factory);
+    }
```

Comment:
```suggestion
    struct returnSimplexSuccessAndTime ret =
        OPT_TryToCallSimplex(options, problemeHebdo, Probleme, NumIntervalle, optimizationNumber,
                optPeriodStringGenerator, true, writer, mps_writer_factory);

    if (!ret.simplexReturnValue)
    {
        ret = OPT_TryToCallSimplex(options, problemeHebdo, Probleme,  NumIntervalle, optimizationNumber,
                optPeriodStringGenerator, false, writer, mps_writer_factory);
    }
```

Reply:
PremierPassage is used later in the function